packages/fiber/src/three-types.ts

```diff
@@ -21,15 +21,16 @@ export type Vector4 = MathType<THREE.Vector4>
 export type Color = MathType<THREE.Color>
 export type Layers = MathType<THREE.Layers>
 export type Quaternion = MathType<THREE.Quaternion>
+export type Euler = MathType<THREE.Euler>
 
-type WithMathProps<P> = { [K in keyof P]: P[K] extends MathRepresentation ? MathType<P[K]> : P[K] }
+type WithMathProps<P> = { [K in keyof P]: P[K] extends MathRepresentation | THREE.Euler ? MathType<P[K]> : P[K] }
 
 interface RaycastableRepresentation {
   raycast(raycaster: THREE.Raycaster, intersects: THREE.Intersection[]): void
 }
 type EventProps<P> = P extends RaycastableRepresentation ? Partial<EventHandlers> : {}
 
-interface ReactProps<P> {
+export interface ReactProps<P> {
```

Comment:
BTW how were you getting `error TS4023: Exported variable {ComponentName} has or is using name 'ReactProps' from external module "C:/dev/nt--drei/node_modules/@react-three/fiber/dist/declarations/src/three-types" but cannot be named.`? Is this with PNPM and symlinks?

Reply:
It is with the Drei repo in the base src folder using yarn and no symlinks (afaik). There must be a better solution than this, but this worked. :)